src/main/java/at/ac/tuwien/kr/alpha/grounder/atoms/IntervalAtom.java

```diff
@@ -1,43 +1,70 @@
 package at.ac.tuwien.kr.alpha.grounder.atoms;
 
+import at.ac.tuwien.kr.alpha.Util;
 import at.ac.tuwien.kr.alpha.common.atoms.Atom;
-import at.ac.tuwien.kr.alpha.common.atoms.Literal;
+import at.ac.tuwien.kr.alpha.common.atoms.FixedInterpretationAtom;
+import at.ac.tuwien.kr.alpha.common.predicates.FixedInterpretationPredicate;
 import at.ac.tuwien.kr.alpha.common.predicates.Predicate;
 import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
 import at.ac.tuwien.kr.alpha.common.terms.IntervalTerm;
 import at.ac.tuwien.kr.alpha.common.terms.Term;
 import at.ac.tuwien.kr.alpha.common.terms.VariableTerm;
 import at.ac.tuwien.kr.alpha.grounder.Substitution;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 
 /**
- * Helper for treating IntervalTerms in rules. Each IntervalTerm is replaced by a variable and a special IntervalAtom
+ * Helper for treating IntervalTerms in rules.
+ *
+ * Each IntervalTerm is replaced by a variable and this special IntervalAtom
  * is added to the rule body for generating all bindings of the variable.
+ *
+ * The first term of this atom is an IntervalTerm while the second term is any Term (if it is a VariableTerm, this will
+ * bind to all elements of the interval, otherwise it is a simple check whether the Term is a ConstantTerm<Integer>
+ * with the Integer being inside the interval.
  * Copyright (c) 2017, the Alpha Team.
  */
-public class IntervalAtom implements Literal {
-	public static final Predicate INTERVAL_PREDICATE = new Predicate("_interval", 2, true);
+public class IntervalAtom extends FixedInterpretationAtom {
+	private static final FixedInterpretationPredicate INTERVAL_PREDICATE = new FixedInterpretationPredicate("_interval", 2, true) {
```

Comment:
Sorry, but I am really confused by this. You anonymously create a `FixedInterpretationPredicate`, which has exactly one public method, and this public method will throw because it is not supported. Please explain.

Reply:
This is why I suspect that we can get rid of the evaluation at the `Predicate`. The `FixedInterpretationPredicate` is probably not really necessary any longer. At an intermediate stage it was needed, but I think it should now work with an ordinary predicate too and this is just cruft.